transforms/shared.js

```diff
@@ -266,6 +269,21 @@ export const handleSpecialCaseOps = (t, knownDecimalNodes, path, opToName) => {
   // create call to Decimal[opToName[operator]]
   let { left, right, operator } = path.node;
 
+  if (
+    operator === "+" &&
+    knownDecimalNodes.has(left) &&
+    knownDecimalNodes.has(right)
+  ) {
+    // Binary + expression with both operands known to be Decimals, like
+    // 0.1m + 0.2m, can be transformed to an .add() method call
```

Comment:
Q: if `left` is a known decimal but `right` isn't, is it still safe to compile it to `.add()`?

`0.1m + x` -> `0.1m.add(x)`.

Reply:
Well spotted. But I think it's not safe? big.js and decimal.js take Numbers, so `Decimal("0.1").add(x)` would succeed if x was a Number, whereas `0.1m + x` would be supposed to throw.

Reply:
Oh yeah it's not safe them, I expected the runtime implementation to have the same constraints as the proposal.